### PR TITLE
Fix/rki tsv csv

### PR DESCRIPTION
**Pull request by rneher**

### CODE REVIEW

**Comment by joverlee521** [comment]
Looks like there were [additional changes to the headers of the file](https://github.com/robert-koch-institut/SARS-CoV-2-Sequenzdaten_aus_Deutschland/issues/49#issuecomment-1584543443) as well. I'll be pushing up more changes shortly.

**Comment by joverlee521** [comment]
Triggered a [test run](https://github.com/nextstrain/ncov-ingest/actions/runs/5468542530) with the latest changes. 

I'll compare the final metadata output from then test run to the current file on S3 when the test run is complete to make sure there are not weird changes.

**Comment by joverlee521** [comment]
The test run completed successfully and uploaded files to the staging S3 bucket. 

Comparing the RKI entries in `s3://nextstrain-staging/files/ncov/open/branch/fix/rki-tsv-csv/metadata.tsv.zst` with RKI entries in the current file at `s3://nextstrain-data/files/ncov/open/metadata.tsv.zst`

```
aws s3 cp s3://nextstrain-staging/files/ncov/open/branch/fix/rki-tsv-csv/metadata.tsv.zst data/rki-test-metadata.tsv.zst
aws s3 cp s3://nextstrain-data/files/ncov/open/metadata.tsv.zst` data/open-metadata.tsv.zst

csvtk filter2 -t -f '$database=="rki"' data/rki-test-metadata.tsv.zst > data/rki-test-rki-only.tsv 
csvtk filter2 -t -f '$database=="rki"' data/open-metadata.tsv.zst > data/open-rki-only.tsv 

csv-diff data/open-rki-only.tsv data/rki-test-rki-only.tsv --key strain > data/rki-only.diff 
```

Summary of changes:
> 633970 rows changed, 7689 rows added, 52473 rows removed

Not much we can do about the removed records, but I took a look at the [first 1000 changed records](https://github.com/nextstrain/ncov-ingest/files/11972778/1000-rki-diff.txt)

Most changes were just the `date_submitted` and `sampling_strategy` fields. There are now a lot of empty `date_submitted` fields but the diff did allow me to catch a new expected date format that I've added in 3a7f3d4fb207e44c11cd96f4a828c0d1f1d0711d. The `sampling_strategy` field changes are just simplifications of the codes that RKI use for sequencing reasons. 

There are some changes to `pango_lineage` but these values come directly from RKI metadata. There are a few changes to `clock_deviation` but I believe that's expected.

File: bin/transform-rki [hunk 1]
```diff
@@ -29,14 +29,12 @@ from lib.utils.transformpipeline.transforms import (AddHardcodedMetadataRki,
...
-    "SEQUENCING_LAB_PC": "submitting_lab",
-    "GISAID_ACCESSION": "strain_gisaid",
-    "lineage": "pango_lineage",
-    "SEQ_REASON": "sampling_strategy",
+    "SEQUENCE.DATE_OF_SAMPLING": "date",
+    "SEQUENCE.PUSHED_TO_DWH": "date_submitted",
+    "DL.ID": "originating_lab",
+    "SL.ID": "submitting_lab",
+    "PANGOLIN.LINEAGE_LATEST": "pango_lineage",
+    "SEQUENCE.SEQUENCING_REASON": "sampling_strategy",
```
Comment (joverlee521) [inline bin/transform-rki:37]: Full disclosure: I made these column changes using Google Translate to translate the [RKI docs of the metadata fields](https://github.com/robert-koch-institut/SARS-CoV-2-Sequenzdaten_aus_Deutschland#variablen-und-werte).

File: bin/fetch-from-rki-metadata [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 set -euo pipefail
 
-curl "https://raw.githubusercontent.com/robert-koch-institut/SARS-CoV-2-Sequenzdaten_aus_Deutschland/master/SARS-CoV-2-Sequenzdaten_Deutschland.csv.xz" \
+curl "https://raw.githubusercontent.com/robert-koch-institut/SARS-CoV-2-Sequenzdaten_aus_Deutschland/master/SARS-CoV-2-Sequenzdaten_Deutschland.tsv.xz" \
```
Comment (corneliusroemer) [inline bin/fetch-from-rki-metadata:4]: Use correct default branch (it's autoforwarded but this is more correct)
```suggestion
curl "https://raw.githubusercontent.com/robert-koch-institut/SARS-CoV-2-Sequenzdaten_aus_Deutschland/main/SARS-CoV-2-Sequenzdaten_Deutschland.tsv.xz" \
```

**Review by corneliusroemer — APPROVED** [review]

File: bin/transform-rki (hunk 1 above)
Comment (corneliusroemer) [reply bin/transform-rki:37]: They are correct :)

**Comment by corneliusroemer** [comment]
I opened an issue in the RKI repo to let them know that most submission dates have gone missing with the new format: https://github.com/robert-koch-institut/SARS-CoV-2-Sequenzdaten_aus_Deutschland/issues/50

File: bin/fetch-from-rki-metadata (hunk 2 above)
Comment (joverlee521) [reply bin/fetch-from-rki-metadata]: Ah, good catch!

Comment (victorlin) [reply bin/fetch-from-rki-metadata]: FYI, in most GitHub URLs, you can `HEAD` in place of the default branch name to be more future-proof. Example:

https://raw.githubusercontent.com/robert-koch-institut/SARS-CoV-2-Sequenzdaten_aus_Deutschland/HEAD/SARS-CoV-2-Sequenzdaten_Deutschland.tsv.xz

This was discussed more in https://github.com/nextstrain/auspice/pull/1660#discussion_r1155025654

**Pull request merged**